src/vector_commitment_of_discrete_log.rs

```diff
@@ -0,0 +1,691 @@
+        .unwrap();
+
+        let generator = secp256k1_group_public_parameters.generator;
+
+        PublicParameters::new::<
+            { secp256k1::SCALAR_LIMBS },
+            secp256k1::Scalar,
+            secp256k1::GroupElement,
+            secp256k1::Scalar,
+            secp256k1::GroupElement,
+            Pedersen<2, { secp256k1::SCALAR_LIMBS }, secp256k1::Scalar, secp256k1::GroupElement>,
+        >(
+            secp256k1_scalar_public_parameters,
+            secp256k1_group_public_parameters,
+            pedersen_public_parameters,
+            [generator; 2],
+        )
+    }
+
+    pub(crate) fn vector_language_public_parameters(
+    ) -> language::PublicParameters<SOUND_PROOFS_REPETITIONS, VectorLang> {
+        let secp256k1_scalar_public_parameters = secp256k1::scalar::PublicParameters::default();
+
+        let secp256k1_group_public_parameters =
+            secp256k1::group_element::PublicParameters::default();
+
+        let multi_pedersen_public_parameters = pedersen::PublicParameters::derive_default::<
+            { secp256k1::SCALAR_LIMBS },
+            secp256k1::GroupElement,
+        >()
+        .unwrap()
+        .into();
+
+        let generator = secp256k1_group_public_parameters.generator;
+
+        PublicParameters::new::<
+            { secp256k1::SCALAR_LIMBS },
+            secp256k1::Scalar,
+            secp256k1::GroupElement,
+            self_product::GroupElement<2, secp256k1::Scalar>,
+            self_product::GroupElement<2, secp256k1::GroupElement>,
+            MultiPedersen<
+                2,
+                { secp256k1::SCALAR_LIMBS },
+                secp256k1::Scalar,
+                secp256k1::GroupElement,
+            >,
+        >(
+            secp256k1_scalar_public_parameters,
+            secp256k1_group_public_parameters,
+            multi_pedersen_public_parameters,
+            [generator; 2],
+        )
+    }
+
+    #[test]
+    fn generates_correct_statement() {
+        let secp256k1_group_public_parameters =
+            secp256k1::group_element::PublicParameters::default();
+
+        let generator = secp256k1::group_element::GroupElement::generator_from_public_parameters(
+            &secp256k1_group_public_parameters,
+        )
+        .unwrap();
+
+        let language_public_parameters = language_public_parameters();
+
+        let witnesses = sample_witnesses::<SOUND_PROOFS_REPETITIONS, Lang>(
+            &language_public_parameters,
+            1,
+            &mut OsRng,
+        );
+
+        let (_, statements) = generate_valid_proof::<SOUND_PROOFS_REPETITIONS, Lang>(
+            &language_public_parameters,
+            witnesses.clone(),
+            &mut OsRng,
+        );
+
+        let [first_message, second_message] = (*witnesses[0].commitment_message()).into();
+
+        assert_eq!(
+            *statements[0].linear_combination_of_discrete_logs(),
+            first_message * generator + second_message * generator
+        );
+    }
```

Comment:
Can we talk about this test? It seems to me you only use a single group element for this test and I think we should have a test where you use multiple group elements.

Reply:
I use two, just the same value. Does it matter? I can change it.